src/Microsoft.AspNet.OData.Shared/RequestPreferenceHelpers.cs

```diff
@@ -33,6 +36,54 @@ internal static bool RequestPrefersReturnNoContent(IWebApiHeaders headers)
             return false;
         }
 
+        internal static bool RequestPrefersMaxPageSize(IWebApiHeaders headers, out int pageSize)
+        {
+            pageSize = -1;
+            IEnumerable<string> preferences = null;
+            if (headers.TryGetValues(PreferHeaderName, out preferences))
+            {
+                pageSize = GetMaxPageSize(preferences, MaxPageSize);
+                if (pageSize >= 0)
+                {
+                    return true;
+                }
+                //maxpagesize gets supersedes odata.maxpagesize
+                pageSize = GetMaxPageSize(preferences, ODataMaxPageSize);
+                if (pageSize >= 0)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private static int GetMaxPageSize(IEnumerable<string> preferences, string preferenceHeaderName)
+        {
+            const int Failed = -1;
+            string maxPageSize = preferences.FirstOrDefault(s => s.Contains(preferenceHeaderName));
```

Comment:
>preferenceHeaderName [](start = 76, length = 20)

Should we do case insensitive comparison here?

Reply:
Thanks for the catch. There is no overload available for Contains that would ignore the case hence using the IndexOf instead of Contains.

Reply:
Should probably apply case-insensitivity to other preferences as well.

---
In reply to: [231698281](https://github.com/OData/WebApi/pull/1673#discussion_r231698281) [](ancestors = 231698281)